src/Orleans.Runtime/Hosting/Generic/SiloHostBuilder.cs

```diff
@@ -33,14 +44,16 @@ public ISiloHost Build()
             // Automatically configure Orleans if it wasn't configured before. 
             // This will not happen once we use the generic host builder from Microsoft.Extensions.Hosting
             this.ConfigureOrleans();
-            
+
             BuildHostConfiguration();
             CreateHostingEnvironment();
             CreateHostBuilderContext();
             BuildAppConfiguration();
 
             var serviceProvider = CreateServiceProvider();
 
+            ValidateSystemConfiguration(serviceProvider);
```

Comment:
When using the generic HostBuilder, would this just be an `IHostedService` that throws in its constructor (instead of StartAsync) or how would it map to an exception during Build?

Reply:
I'm not sure what the best approach is. We need some kind of lifecycle hook that we can put this and similar procedures in.